File: java/83remove-duplicates-from-sorted-list/83.删除排序链表中的重复元素.java
/*
 * @lc app=leetcode.cn id=83 lang=java
 *
 * [83] 删除排序链表中的重复元素
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head == null || head.next == null) {
            return head;
        }
        ListNode lastNode = head;
        ListNode node = head.next;
        while(node != null) {
            if(node.val == lastNode.val) {
                lastNode.next = node.next;
            } else {
                lastNode = node;
            }
            node = node.next;
        }
        return head;
    }
}
// @lc code=end

File: go/83remove-duplicates-from-sorted-list/83.删除排序链表中的重复元素.go
/*
 * @lc app=leetcode.cn id=83 lang=golang
 *
 * [83] 删除排序链表中的重复元素
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
	if head == nil || head.Next == nil {
		return head
	}
	node := head.Next
	lastNode := head
	for node != nil {
		if lastNode.Val == node.Val {
			lastNode.Next = node.Next
		} else {
			lastNode = node
		}
		node = node.Next
	}
	return head
}
// @lc code=end

